topdown/bytes.go

```diff
@@ -0,0 +1,95 @@
+package topdown
+
+import (
+	"fmt"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/topdown/builtins"
+	"math/big"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type multiplier int
+
+const (
+	base = 1024
+
+	none multiplier = 1
+	kb              = base
+	mb              = kb * base
+	gb              = mb * base
+	tb              = gb * base
+)
+
+func numBytesErr(msg string) error {
+	return fmt.Errorf("numbytes error: %s", msg)
+}
+
+var (
+	errNoAmount = numBytesErr("no byte amount provided")
+	errIntConv  = numBytesErr("could not parse amount to integer")
+)
+
+func errUnitNotRecognized(unit string) error {
+	return numBytesErr(fmt.Sprintf("unit %s not recognized", unit))
+}
+
+func builtinNumBytes(a ast.Value) (ast.Value, error) {
+	var m multiplier
+
+	numPattern := `([0-9]+)`
+	strPattern := `([a-z]+)`
+
+	raw, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	s := strings.ToLower(raw.String())
+
+	numRe, err := regexp.Compile(numPattern)
+	if err != nil {
+		return nil, err
+	}
+
+	strRe, err := regexp.Compile(strPattern)
+	if err != nil {
+		return nil, err
+	}
+
+	numStr := numRe.FindString(s)
+	unit := strRe.FindString(s)
+
+	if numStr == "" {
+		return nil, errNoAmount
+	}
+
+	num, err := strconv.Atoi(numStr)
```

Comment:
We shouldn't use `strconv.Atoi` and `int` types since we can run into problems later on when we multiply by the units. Lets switch to using https://golang.org/pkg/math/big/ with either ints for (if we want to support decimal places) floats and use their string parse helper.